### ISSUE TITLE
Optimize Hyperparameters of RandomForestClassifier

### ISSUE THREAD

**Issue by DeepBlockDeepak** [DeepBlockDeepak/kaggle_titanic#8]
Try grid search or random search in new file:  `src/random_forest/rf_hyperparameter_tuning.py`

**Comment by DeepBlockDeepak** [comment]
> Best parameters: {'bootstrap': True, 'criterion': 'entropy', 'max_depth': 10, 'max_features': 'sqrt', 'min_samples_leaf': 1, 'min_samples_split': 2, 'n_estimators': 50}
Best cross-validation score: 0.8455727371220328
Validation accuracy of the best model: 0.7877094972067039
